services/QuillLMS/app/workers/populate_evidence_activity_healths_worker.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class PopulateEvidenceActivityHealthsWorker
+  include Sidekiq::Worker
+
+  INTERVAL = 180 # 3 minutes
+
+  def perform
+    Evidence::ActivityHealth.destroy_all
+    # execute this command to start primary key indices back from 1
+    ActiveRecord::Base.connection.execute("TRUNCATE evidence_activity_healths RESTART IDENTITY CASCADE")
+
+    # spread these, to cut down on DB resource contention.
+    Evidence::Activity.all.each.with_index do |activity, index|
+      PopulateEvidenceActivityHealthWorker.perform_in(index * INTERVAL, activity.id)
```

Comment:
Just a note on this in case it hasn't been considered: if something goes wrong with the `PopulateEvidenceActivityHealthWorker` calls, we could end up truncating the old health data and then failing to re-populate it.  Given the nature of us running this every night, that might not be a big problem, but it's worth confirming that this makes sense to us.

Reply:
I'm going to say that's okay for now, since this data is updated frequently enough that we could fix the issue quickly. Also the individual stats are accessible from other places on the admin panel, so if this table is somehow truncated accidentally admin will still be able to view basic stats elsewhere.